### PR TITLE
More robust 'grakn server status' check

### PR DESCRIPTION
**Pull request by lolski**
There were two occasions where `ps ... | grep ...` used for engine and redis uptime check would give wrong result. For example, if the working directory is also named `Grakn`, or `redis-server`. 

This PR adds a fix to protect against these edge cases.

In addition to that, the logic for the uptime checks and getting the proper redis binary name are now extracted into their own dedicated functions, in order to reduce cognitive overload to human readers.

Blergh, so much engineering effort spent just for uptime check. I am open to suggestion if there are better approach to ps | grep

### CODE REVIEW

File: grakn-dist/src/grakn [hunk 1]
```diff
@@ -68,6 +70,15 @@ update_classpath_global_var() {
   CLASSPATH="$CLASSPATH":"${GRAKN_HOME}"/services/grakn
 }
 
+ps_ef_grep_processname() {
+  ps -ef | grep -v grep | awk '{print $2" "$(NF-1)}' | grep "$1"
```
Comment (burukuru) [inline grakn-dist/src/grakn]: I'd suggest pgrep if that works on mac

File: grakn-dist/src/grakn [hunk 2]
```diff
@@ -68,6 +70,15 @@ update_classpath_global_var() {
...
+}
+
+
+ps_ef_grep_processname_by_java_classname() {
+  ps -ef | grep -v grep | awk '{print $2" "$(NF)}' | grep "$1"
```
Comment (burukuru) [inline grakn-dist/src/grakn]: `jps` should give you what you want

Comment (lolski) [reply grakn-dist/src/grakn]: @burukuru as far as I know, `jps` isn't part of OpenJDK which we claim to support

Comment (burukuru) [reply grakn-dist/src/grakn]: oh that sucks

File: grakn-dist/src/grakn (hunk 1 above)
Comment (burukuru) [reply grakn-dist/src/grakn]: apparently mac doesn't have pgrep so nevermind..

File: grakn-dist/src/grakn (hunk 2 above)
Comment (lolski) [reply grakn-dist/src/grakn]: yup. the joy of maintaining platform compatibility...

**Comment by aelred** [comment]
Is this ready for review/merging? I don't know if @burukuru wants to request changes.

**Pull request merged**